adapters/inmobi/inmobi.go

```diff
@@ -0,0 +1,97 @@
+package inmobi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type InmobiAdapter struct {
+	endPoint string
+}
+
+func NewInmobiAdapter(endpoint string) *InmobiAdapter {
+	return &InmobiAdapter{
+		endPoint: endpoint,
+	}
+}
+
+func (a *InmobiAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	for _, imp := range request.Imp {
+		if imp.Banner != nil {
+			banner := imp.Banner
```

Comment:
The banner is shared memory. You need to make a copy before editing it. That code looks like:
```
banner := *imp[i].Banner
imp[i].Banner = &banner
...
```